app/main.py

```diff
@@ -1,34 +1,176 @@
+from __future__ import annotations
+
+
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(
+            self,
+            row: int,
+            column: int,
+            is_alive: bool = True
+    ) -> None:
+
+        self.row = row
+        self.column = column
+        self.is_alive = is_alive
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
-
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
-
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
-
-
-class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+
+        self.len_ship = round(((start[0] - end[0]
+                                )**2 + (start[1] - end[1])**2)**0.5) + 1
+        self.ship = [Deck(end[0], end[1] - i)
+                     if start[1] < end[1]
+                     else Deck(end[0] - i, end[1])
+                     for i in range(self.len_ship)]
+        self.is_drowned = is_drowned
+
+    def get_deck(
+        self,
+        row: int,
+        column: int
+    ) -> Deck | Ship:
+
+        for ship in self.ships:
```

Comment:
I see only `self.ship` in `__init__()` method. Where the `self.ships` is created?